pkg/auth/signatures.go

```diff
@@ -0,0 +1,67 @@
+package auth
+
+import (
+	"io"
+	"os"
+
+	"github.com/square/p2/pkg/util"
+	"golang.org/x/crypto/openpgp"
+)
+
+// Something that can verify PGP signatures
+type SigVerifier interface {
+	CheckDetachedSignature(signed, signature io.Reader) (*openpgp.Entity, error)
+}
+
+// A SignatureVerifier that reads its keyring from a file. The file is
+// read every time it is needed.
```

Comment:
I though manifests were verified only when their pod was starting.  That does change things :)